File: 1677-MatrixDiagonalSum/1677-MatrixDiagonalSum.cpp
// Last updated: 7/9/2026, 3:06:04 PM
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int n=mat.size();
        int sum=0;
        for(int i =0; i<n;i++){
            sum+= mat[i][i]+ mat[i][n-i-1];
        }
        if(n%2==0){
            return sum;
        }
        else{
            return sum -= mat[n/2][n/2];
        }
    }
};

File: 1677-MatrixDiagonalSum/1677-MatrixDiagonalSum.java
// Last updated: 7/9/2026, 3:05:56 PM
class Solution {
    public int diagonalSum(int[][] mat) {
        int n=mat.length;
        int r=0;
        for(int i=0;i<n;i++){
            r+=mat[i][i] + mat[i][n-i-1];

        }
        if(n%2==1){
            r-=mat[n/2][n/2];
        }
        return r;


        
    }
}